crates/containerd-shim-wasm/src/sandbox/cgroups/cgroupv2.rs

```diff
@@ -26,16 +27,35 @@ impl CgroupV2 {
 
     // Make sure the cgroup parents exist and that cgroup controllers are delegated from the parent cgroup.
     fn ensure_parents(&self) -> Result<()> {
+        debug!("ensuring cgroup parents exist: {}", self.path.display());
         let mut full = safe_join(PathBuf::from("/"), self.base.clone())?;
 
-        for d in self.path.clone().iter() {
+        let p = self.path.clone();
+        for d in p.iter() {
             full = safe_join(full.clone(), PathBuf::from(d))?;
 
             if full.exists() {
+                debug!("skipping creation of existing cgroup: {}", full.display());
                 // This is a pre-existing cgroup, so we won't touch subtree control.
                 continue;
             }
 
+            if full.to_str() == safe_join(self.base.clone(), self.path.clone())?.to_str() {
```

Comment:
Is this the only place that the full path (base + path) is calcuated? might consider making base+path a method on the implementation if reused somewhere